### PR TITLE
[Hub Generated] Review request for Microsoft.VnfManager to add version preview/2020-01-01-preview

### PR DESCRIPTION
**Pull request by neelesh06**
If you are a MSFT employee you can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/neelesh06/azure-rest-api-specs/dev-vnfmanager-Microsoft.VnfManager-2020-01-01-preview...master/).

### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/vnfmanager/resource-manager/Microsoft.VnfManager/preview/2020-01-01-preview/device.json [hunk 1]
```diff
@@ -0,0 +1,489 @@
...
+        "mecRegistrationKey": {
+          "type": "string",
+          "description": "The registration key for the Vnf Manager device resource"
+        }
+      }
+    },
+    "DeviceRegistrationKey": {
+      "description": "Vnf Manager Device registration key",
+      "properties": {
+        "registrationKey": {
```
Comment (sanjaiganesh) [inline specification/vnfmanager/resource-manager/Microsoft.VnfManager/preview/2020-01-01-preview/device.json]: >registrationKey [](start = 9, length = 15)

annotate with x-ms-secret: true

Comment (sanjaiganesh) [reply specification/vnfmanager/resource-manager/Microsoft.VnfManager/preview/2020-01-01-preview/device.json]: is this same as mecRegistrationKey in PUT.. ? probably use same name in both ?

---
In reply to: [367744014](https://github.com/Azure/azure-rest-api-specs/pull/8162#discussion_r367744014) [](ancestors = 367744014)

File: specification/vnfmanager/resource-manager/Microsoft.VnfManager/preview/2020-01-01-preview/device.json [hunk 2]
```diff
@@ -0,0 +1,489 @@
...
+          "description": "The sku of the Vnf Manager device"
+        },
+        "mecVNFs": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/VNFRef"
+          },
+          "description": "The list of VNFs deployed on the Vnf Manager device."
+        },
+        "mecRegistrationKey": {
```
Comment (sanjaiganesh) [inline specification/vnfmanager/resource-manager/Microsoft.VnfManager/preview/2020-01-01-preview/device.json]: >mecRegistrationKey [](start = 9, length = 18)

annotate with x-ms-secret: true

File: specification/vnfmanager/resource-manager/Microsoft.VnfManager/preview/2020-01-01-preview/vendor.json [hunk 3]
```diff
@@ -0,0 +1,859 @@
...
+          "final-state-via": "azure-async-operation"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.VnfManager/vendors": {
+      "get": {
+        "tags": [
+          "Vendor"
+        ],
+        "operationId": "Vendors_ListAllBySubscription",
```
Comment (sanjaiganesh) [inline specification/vnfmanager/resource-manager/Microsoft.VnfManager/preview/2020-01-01-preview/vendor.json]: >Vendors_ListAllBySubscription [](start = 24, length = 29)

Vendors_ListBySubscription

File: specification/vnfmanager/resource-manager/Microsoft.VnfManager/preview/2020-01-01-preview/device.json (hunk 1 above)
Comment (neelesh06) [reply specification/vnfmanager/resource-manager/Microsoft.VnfManager/preview/2020-01-01-preview/device.json]: Yes. I have changed and replace the same name.

File: specification/vnfmanager/resource-manager/Microsoft.VnfManager/preview/2020-01-01-preview/vendor.json (hunk 3 above)
Comment (neelesh06) [reply specification/vnfmanager/resource-manager/Microsoft.VnfManager/preview/2020-01-01-preview/vendor.json]: changed.

File: specification/vnfmanager/resource-manager/Microsoft.VnfManager/preview/2020-01-01-preview/device.json (hunk 2 above)
Comment (neelesh06) [reply specification/vnfmanager/resource-manager/Microsoft.VnfManager/preview/2020-01-01-preview/device.json]: I am getting the Schema Violation when I am using the property "x-ms-secret".
ERROR: Schema violation: Additional properties not allowed: x-ms-secret
    - file:///C:/Users/neeleshk/azure-rest-api-specs/specification/vnfmanager/resource-manager/Microsoft.VnfManager/preview/2020-01-01-preview/device.json:458:8 ($.definitions.DeviceRegistrationKey.properties.mecRegistrationKey)

I have not added this property but fixed other comments.

File: specification/vnfmanager/resource-manager/Microsoft.VnfManager/preview/2020-01-01-preview/device.json (hunk 1 above)
Comment (neelesh06) [reply specification/vnfmanager/resource-manager/Microsoft.VnfManager/preview/2020-01-01-preview/device.json]: I am getting the Schema Violation when I am using the property "x-ms-secret".
ERROR: Schema violation: Additional properties not allowed: x-ms-secret
    - file:///C:/Users/neeleshk/azure-rest-api-specs/specification/vnfmanager/resource-manager/Microsoft.VnfManager/preview/2020-01-01-preview/device.json:458:8 ($.definitions.DeviceRegistrationKey.properties.mecRegistrationKey)

I have not added this property but fixed other comments.

File: specification/vnfmanager/resource-manager/Microsoft.VnfManager/preview/2020-01-01-preview/device.json [hunk 4]
```diff
@@ -0,0 +1,489 @@
...
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.VnfManager/devices": {
+      "get": {
+        "tags": [
+          "VnfManagerDevices"
+        ],
+        "operationId": "VnfManagerDevices_ListAll",
```
Comment (sanjaiganesh) [inline specification/vnfmanager/resource-manager/Microsoft.VnfManager/preview/2020-01-01-preview/device.json]: >VnfManagerDevices_ListAll [](start = 24, length = 25)

VnfManagerDevices_ListBySubscription

File: specification/vnfmanager/resource-manager/Microsoft.VnfManager/preview/2020-01-01-preview/vnf.json [hunk 5]
```diff
@@ -0,0 +1,431 @@
...
+        },
+        "sku": {
+          "type": "string",
+          "description": "The sku for the Vnf Manager Vnf."
+        },
+        "vendor": {
+          "type": "string",
+          "description": "The vendor for the Vnf Manager Vnf."
+        },
+        "vnfServiceKey": {
```
Comment (sanjaiganesh) [inline specification/vnfmanager/resource-manager/Microsoft.VnfManager/preview/2020-01-01-preview/vnf.json]: >vnfServiceKey [](start = 9, length = 13)

if it is secret, annotate it with x-ms-secret:true

File: specification/vnfmanager/resource-manager/Microsoft.VnfManager/preview/2020-01-01-preview/vendor.json [hunk 6]
```diff
@@ -0,0 +1,867 @@
...
+          "Get all the skus of Vendor resource": {
+            "$ref": "./examples/VendorSkuListAll.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.VnfManager/locations/{locationName}/vendors/{vendorName}/vnfs/{vnfServiceKey}": {
```
Comment (sanjaiganesh) [inline specification/vnfmanager/resource-manager/Microsoft.VnfManager/preview/2020-01-01-preview/vendor.json]: >/subscriptions/{s [](start = 5, length = 17)

for all tracked resources, define 'patch' that allows updating at least the tags

File: specification/vnfmanager/resource-manager/Microsoft.VnfManager/preview/2020-01-01-preview/device.json (hunk 4 above)
Comment (neelesh06) [reply specification/vnfmanager/resource-manager/Microsoft.VnfManager/preview/2020-01-01-preview/device.json]: done

File: specification/vnfmanager/resource-manager/Microsoft.VnfManager/preview/2020-01-01-preview/vnf.json (hunk 5 above)
Comment (neelesh06) [reply specification/vnfmanager/resource-manager/Microsoft.VnfManager/preview/2020-01-01-preview/vnf.json]: I am getting the Schema Violation when I am using the property "x-ms-secret".
ERROR: Schema violation: Additional properties not allowed: x-ms-secret
- file:///C:/Users/neeleshk/azure-rest-api-specs/specification/vnfmanager/resource-manager/Microsoft.VnfManager/preview/2020-01-01-preview/device.json:458:8 ($.definitions.DeviceRegistrationKey.properties.mecRegistrationKey)
I have not added this property but fixed other comments.

File: specification/vnfmanager/resource-manager/Microsoft.VnfManager/preview/2020-01-01-preview/vendor.json (hunk 6 above)
Comment (neelesh06) [reply specification/vnfmanager/resource-manager/Microsoft.VnfManager/preview/2020-01-01-preview/vendor.json]: Vendor and sku are the proxy resource. these does not need any tags update.

**Review by njuCZ — APPROVED** [review]

**Comment by njuCZ** [comment]
@neelesh06 please fix the conflicts

**Comment by njuCZ** [comment]
@azuresdkci this PR has conflicts, It must be fixed before I merge it

**Comment by njuCZ** [comment]
@neelesh06 it seems long time no response for this PR, Could you please fix the conflicts so I could merge it ? If not, I am going to close it

**Comment by neelesh06** [comment]
closing this PR as this is no longer needed,

**Comment by neelesh06** [comment]
I closed the PR.

**Pull request closed without merging**